pkg/controller/storagecluster/reconcile.go

```diff
@@ -924,6 +926,15 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string, nodeCount int, r
 	} else {
 		reqLogger.Info(fmt.Sprintf("No monDataDirHostPath, monPVCTemplate or storageDeviceSets configured for storageCluster %s", sc.GetName()))
 	}
+	multusNetwork := sc.Spec.Network
+	multusNetworkNil := ocsv1.NetworkSpec{}
+	if multusNetwork != multusNetworkNil {
+		multusSelector := make(map[string]string)
+		cephCluster.Spec.Network.Provider = networkProvider
```

Comment:
Instead of assigning individually, what about doing something like: `cephCluster.Spec.Network = ocsv1.NetworkSpec`? If the structure is identical we should be able to just pass it to Rook, right?

Reply:
`cephCluster.Spec.Network` and `ocsv1.NetworkSpec` doesn't have identical structure so can't pass it directly to rook.

Reply:
Why can't it be identical?

Reply:
I wanted to enforce `Public` and `Cluster` instead of the selectors as they are in Rook
I wanted to hardcode `multus` for the time being so not making `cephCluster.Spec.Network` and `ocsv1.NetworkSpec` identical

This is the Rook Network Spec
```go
type NetworkSpec struct {
	Provider  string `json:"provider"`
	Selectors map[string]string `json:"selectors"`
}
```

This is the one for OCS
```go
type NetworkSpec struct {
    Public  string `json:"public"`
    Cluster string `json:"cluster"`
}
```
So for the OCS spec the user is bound to specify Public and Cluster.

If this does not make sense then I will make both the spec same.

Reply:
Made the changes